konfig

```diff
@@ -85,27 +86,42 @@ merge() {
 
 import_ctx() {
     local tmpcfg
+    local tmpinputcfg=""
     local out=""
     local arg=""
-    if [[ "$1" == '-p' || "$1" == '--preserve-structure' ]]; then
-      arg="$1"
-      shift
-    fi
-    if [[ "$1" = '--save' || "$1" = '-s' ]]; then
-      out="${XDG_CACHE_HOME:-$HOME/.kube}/config"
-      shift
-    elif [[ "$1" =~ ^-(.*) ]]; then
-      error "unrecognized flag \"$1\""
-    fi
+    for OPT in "$@"
+    do
+      case $OPT in
+        -p | --preserve-structure)
+          arg="$1"
+          shift
+          ;;
+        -s | --save)
+          out="${XDG_CACHE_HOME:-$HOME/.kube}/config"
+          shift
+          ;;
+        -i)
+          if [[ -p /dev/stdin ]]; then
+          tmpinputcfg=$(mktemp konfig_input_XXXXXX)
+            TMPFILES+=( "$tmpinputcfg" )
+            cat - > "$tmpinputcfg"
+            shift
+          fi
+          ;;
+        -*)
+          error "unrecognized flag \"$1\""
+          ;;
+      esac
+    done
     tmpcfg=$(mktemp konfig_XXXXXX)
     TMPFILES+=( "$tmpcfg" )
     $KUBECTL config view --raw > "$tmpcfg"
 
     if [[ -z "$out" ]]; then
-      merge "$arg" "$tmpcfg" "$@"
+      merge "$arg" "$tmpcfg" "$tmpinputcfg" "$@"
```

Comment:
This is not optimal, because it will lead to a trailing `:` in the KUBECONFIG var. Consider this:

```sh
bla() {
	IFS=$':\n\t'
	echo "$*"
}

bla "first" "second"
bla "first" ""
```

which prints

> first:second
> first:

Ideally, we want to have

> first:second
> first


---

Instead, I'd recommend to work with arrays. Consider

```sh
bla() {
	IFS=$':\n\t'
	echo "$*"
}

WORDS=()

WORDS+=("first")
bla "${WORDS[@]}"

WORDS+=("second")
bla "${WORDS[@]}"
```

which prints

> first
> first:second

Reply:
Changed to use array. Thank you for the comment.